### PR TITLE
Use new mdx-deck api

### PR DESCRIPTION
**Pull request by pomber**
mdx-deck API is changing  
See jxnblk/mdx-deck#144

### CODE REVIEW

File: packages/mdx-deck-code-surfer/src/deck-code-surfer.js [hunk 1]
```diff
@@ -1,93 +1,76 @@
...
+      super(props);
+      const { update, index } = props.deck;
+      const steps = props.steps ? props.steps.length : 0;
+      update(setSteps(index, steps));
+    }
 
-      componentWillUnmount() {
-        document.body.removeEventListener("keydown", this.handleKeyDown);
-      }
+    shouldComponentUpdate() {
```
Comment (jxnblk) [inline packages/mdx-deck-code-surfer/src/deck-code-surfer.js]: Not 100% sure the intent here, but should this be using the `nextProps` argument for checking?

File: packages/mdx-deck-code-surfer/src/deck-code-surfer.js [hunk 2]
```diff
@@ -1,93 +1,76 @@
 import CodeSurfer from "code-surfer";
 import React from "react";
-import { modes, incStep, decStep } from "mdx-deck";
-import { withDeck } from "mdx-deck/dist/context";
-import { withSlide } from "mdx-deck/dist/Slide";
+import { withDeck, withSlide } from "mdx-deck";
+import { setSteps } from "mdx-deck/dist/updaters";
+import { modes } from "mdx-deck/dist/constants";
```
Comment (jxnblk) [inline packages/mdx-deck-code-surfer/src/deck-code-surfer.js]: Looks like you don't need to use the `modes` constants anymore – ideally anything you need should be available without digging into the `dist/` path. I can expose the updaters as part of the public API as well

File: packages/mdx-deck-code-surfer/src/deck-code-surfer.js (hunk 1 above)
Comment (pomber) [reply packages/mdx-deck-code-surfer/src/deck-code-surfer.js]: You are right, thanks!

File: packages/mdx-deck-code-surfer/src/deck-code-surfer.js (hunk 2 above)
Comment (pomber) [reply packages/mdx-deck-code-surfer/src/deck-code-surfer.js]: I'm not using `modes` right now, but I'll probably need some way of knowing which one is the current mode later.

Comment (jxnblk) [reply packages/mdx-deck-code-surfer/src/deck-code-surfer.js]: yeah, I'm thinking internally there should be a way to know whether components like this are "live" (i.e. being presented) vs displaying all contents, such as when exporting to PDF. I can expose the updaters and constants from the main `index.js` for now, but would also like to have a simpler API for third party components to be able to hook into

**Pull request merged**